test/e2e/util/util.go

```diff
@@ -135,8 +135,19 @@ func WaitForGuestClient(t *testing.T, ctx context.Context, client crclient.Clien
 	return guestClient
 }
 
-func WaitForNReadyNodes(t *testing.T, ctx context.Context, client crclient.Client, n int32, platform hyperv1.PlatformType) []corev1.Node {
+func WaitForNReadyNodes(t *testing.T, ctx context.Context, client crclient.Client, n int32, platform hyperv1.PlatformType, nodePools ...string) []corev1.Node {
```

Comment:
this is changing the signature but never making use of it? To prove the change it's needed this PR will need to include how this is consumed?

Reply:
I've adjusted the signature, but it's an "optional" parameter, which means that other functions does not need to include into the function call.

Putting this param in this way, we make sure that:
- The change, does not affect other implementations in other tests (you can call this functions without the last parameter.
- You can filter by NodePool, or not, using the same function

For now I'm only using the first index member in the slice, because it will add another loop level in the waitPoll part. I've though to revisit when we need to filter by more than one NodePool.

Reply:
I will reimplement it in other function, makes totally sense the review :)